packages/flash-swap/test/shared/fixtures.ts

```diff
@@ -70,15 +72,23 @@ export async function integrationFixture(signers: Signer[]): Promise<Integration
     ])
   );
 
-  const godModeUniswapV2PairArtifact: Artifact = await artifacts.readArtifact("GodModeUniswapV2Pair");
-  const uniswapV2Pair: GodModeUniswapV2Pair = <GodModeUniswapV2Pair>(
-    await deployContract(deployer, godModeUniswapV2PairArtifact, [])
+  const uniswapV2FactoryArtifact: Artifact = await artifacts.readArtifact("UniswapV2Factory");
+  const uniswapV2Factory: UniswapV2Factory = <UniswapV2Factory>(
+    await deployContract(deployer, uniswapV2FactoryArtifact, [await deployer.getAddress()])
   );
-  await uniswapV2Pair.initialize(wbtc.address, usdc.address);
+  await uniswapV2Factory.createPair(usdc.address, wbtc.address);
+  const pairAddress = await uniswapV2Factory.allPairs(0);
+
+  const uniswapV2PairArtifact: Artifact = await artifacts.readArtifact("UniswapV2Pair");
+  const uniswapV2Pair = new Contract(pairAddress, uniswapV2PairArtifact.abi, deployer) as UniswapV2Pair;
 
   const hifiFlashUniswapV2Artifact: Artifact = await artifacts.readArtifact("HifiFlashUniswapV2");
   const hifiFlashUniswapV2: HifiFlashUniswapV2 = <HifiFlashUniswapV2>(
-    await deployContract(deployer, hifiFlashUniswapV2Artifact, [balanceSheet.address, [uniswapV2Pair.address]])
+    await deployContract(deployer, hifiFlashUniswapV2Artifact, [
+      balanceSheet.address,
+      uniswapV2Factory.address,
+      utils.keccak256(uniswapV2PairArtifact.bytecode),
```

Comment:
This has to be double-checked that it is uses the correct `keccak256` function. Ethers offers two versions:

1. The one from the `@ethersproject/keccak256` package
2. The other from the `@ethersproject/solidity` package.

Reply:
**Update**: it seems like they both return the same value.

```ts
const foo: string = keccak256(uniswapV2PairArtifact.bytecode);
console.log({ foo });

const bar: string = solidityKeccak256(["bytes"], [uniswapV2PairArtifact.bytecode]);
console.log({ bar });
```

Both `foo` and `bar` are 0xc52408a0342ead8a645a7d06aee76aa26cee596d691ed85e2103f71b9cd8d6d0.